packages/amazonq/src/app/inline/recommendationService.ts

```diff
@@ -119,7 +122,41 @@ export class RecommendationService {
             })
             const t0 = performance.now()
 
-            const result = await this.getRecommendationsWithTimeout(languageClient, request, token)
+            // Yield event loop to let the document listen catch updates
+            await sleep(1)
+            // Best effort estimate of deletion
+            const isTriggerByDeletion = documentEventListener.isLastEventDeletion(document.uri.fsPath)
+
+            const ps: Promise<InlineCompletionListWithReferences>[] = []
+            /**
+             * IsTriggerByDeletion is to prevent user deletion invoking Completions.
+             * PartialResultToken is not a hack for now since only Edits suggestion use partialResultToken across different calls of [getAllRecommendations],
+             * Completions use PartialResultToken with single 1 call of [getAllRecommendations].
+             * Edits leverage partialResultToken to achieve EditStreak such that clients can pull all continuous suggestions generated by the model within 1 EOS block.
+             */
+            if (!isTriggerByDeletion && !request.partialResultToken) {
+                const completionPromise: Promise<InlineCompletionListWithReferences> = languageClient.sendRequest(
+                    inlineCompletionWithReferencesRequestType.method,
+                    request,
+                    token
+                )
+                ps.push(completionPromise)
+            }
+
+            /**
+             * Though Edit request is sent on keystrokes everytime, the language server will execute the request in a debounced manner so that it won't be immediately executed.
+             */
+            const editPromise: Promise<InlineCompletionListWithReferences> = languageClient.sendRequest(
+                editCompletionRequestType.method,
+                request,
+                token
+            )
+            ps.push(editPromise)
+
+            let result = await Promise.race(ps)
```

Comment:
`editPromise` will likely always win the race because for most users the edit predictions are not enabled, hence its server API response will be very fast, and it will win the race.

Reply:
donee